pkg/beacon/beacon.go

```diff
@@ -102,6 +104,16 @@ func Initialize(
 					request.BlockNumber,
 				)
 			}()
+		} else {
+			groupPublicKey, err := altbn128.UnmarshalG2Point(request.GroupPublicKey)
+			if err != nil {
+				logger.Warningf("could not relay messages: [%v]", err)
+			} else {
+				// Channel name used for relay entry request is the string
+				// representation of the compressed group public key G2 point.
+				name := hex.EncodeToString(groupPublicKey.Compress())
```

Comment:
Instead of defining `UnmarshalG2Point` in `altbn128.sol` and having a part of the channel naming logic here, we can extract everything to `node.go`:
```
// channelNameForPublicKey takes group public key represented by G2 point
// and transforms it into a broadcast channel name.
// Broadcast channel name for group is the hexadecimal representation of
// compressed public key of the group.
func channelNameForPublicKey(groupPublicKey *bn256.G2) string {
	altbn128GroupPublicKey := altbn128.G2Point{G2: groupPublicKey}
	return hex.EncodeToString(altbn128GroupPublicKey.Compress())
}

// channelNameForPublicKey takes group public key represented by marshalled
// G2 point and transforms it into a broadcast channel name.
// Broadcast channel name for group is the hexadecimal representation of
// compressed public key of the group.
func channelNameForPublicKeyBytes(groupPublicKey []byte) (string, error) {
	g2 := new(bn256.G2)

	if _, err := g2.Unmarshal(groupPublicKey); err != nil {
		return "", fmt.Errorf("could not create channel name: [%v]", err)
	}

	return channelNameForPublicKey(g2), nil
}
```

and then, this part can be just:
```
} else {
  node.ForwardSignatureShares(request.GroupPublicKey)
}
```